code/modules/mob/living/simple_animal/abnormality/he/fan.dm

```diff
@@ -56,3 +65,110 @@
 	else
 		safe+=user
 		to_chat(user, "<span class='nicegreen'>You could use some more.</span>")
+
+//Meltdown
+/mob/living/simple_animal/hostile/abnormality/fan/AttemptWork(mob/living/carbon/human/user, work_type)
+	if(turned_off)
+		to_chat(user, "<span class='nicegreen'>You hit the on switch. Aaah, that feels nice.</span>")
+		TurnOn()
+		return FALSE
+	if(datum_reference.console.meltdown)
+		safework = TRUE
+	return ..()
+
+/mob/living/simple_animal/hostile/abnormality/fan/funpet(mob/petter)
+	if(turned_off)
+		to_chat(petter, "<span class='nicegreen'>You hit the on switch. Aaah, that feels nice.</span>")
+		TurnOn()
+
+//Breach
+/mob/living/simple_animal/hostile/abnormality/fan/ZeroQliphoth(mob/living/carbon/human/user)
+	. = ..()
+	if(!turned_off)
+		TurnOff()
+
+/mob/living/simple_animal/hostile/abnormality/fan/proc/TurnOn(mob/living/carbon/human/user)
+	turned_off = FALSE
+	icon_state = "fan"
+	desc = "It appears to be an office fan."
+	for(var/mob/living/carbon/human/L in GLOB.player_list)
+		if(z != L.z || L.stat >= HARD_CRIT) // on a different Z level or dead
+			continue
+		var/datum/status_effect/stacking/fanhot/V = L.has_status_effect(/datum/status_effect/stacking/fanhot)
+		if(!V)
+			continue
+		else
+			V.Destroy()
+
+/mob/living/simple_animal/hostile/abnormality/fan/proc/TurnOff(mob/living/carbon/human/user)
+	turned_off = TRUE
+	icon_state = "fan_idle"
+	desc = "It appears to be an office fan. It looks like it's turned off."
+	HeatWave()
+
+/mob/living/simple_animal/hostile/abnormality/fan/proc/HeatWave(started)
+	if(!turned_off)
+		datum_reference.qliphoth_change(1)
+		return
+	if(!started)
+		sound_to_playing_players_on_level('sound/abnormalities/seasons/summer_idle.ogg', 150, zlevel = z)
```

Comment:
That's the only place the proc's argument is used. Can't you just move it to the place where you call HeatWave for the first time?